### PR TITLE
[ssreflect] Respect Opaque in FO unification

### PR DESCRIPTION
**Pull request by maximedenes**
This is only a very partial fix, since ssr uses evarconv which doesn't respect Opaque directives.

- [x] CI passed (unrelated timeout in VST)

FTR: This PR is a new incarnation of #7021 that was closed/recreated to use `CODEOWNERS`

### CODE REVIEW

**Review by gares — APPROVED** [review]

**Comment by Zimmi48** [comment]
The VST failure was not spurious: it is currently broken with master (but it was the case before this PR). The first time it broke was https://circleci.com/workflow-run/4e88330f-7e00-4e59-8b64-3e37ba371ba8 (new merging process) so obviously the problem is with VST. Someone should report the problem upstream and encourage them to test their code with master before pushing (and highlight that this is a requirement documented in https://github.com/coq/coq/tree/master/dev/ci#information-for-external-library--plugin-authors to be part of our CI). If no one does it before me, I'll do it but I'm a bit busy until tomorrow afternoon.

**Comment by ejgallego** [comment]
Well I saw this failure before but it seems like a timeout. More testing may be needed.

**Comment by herbelin** [comment]
Let's start to do things properly. On the procedure discussed #7014, we agreed that explicit time was left to ensure that discussion happened. It it not that I have something particular to say about this PR but to ensure that a discussion converges, there is the prerequisite that a discussion has started.

I'm not asking a lot, simply a formal sentence of the form: "is there any discussion wanted on this PR. If not, I shall merge it tomorrow evening". Otherwise, we shall have the worse of #5986.

**Comment by Zimmi48** [comment]
@ejgallego It is a repeated timeout that happened in *all* recent tests both on CircleCI and Travis.

**Comment by maximedenes** [comment]
> I'm not asking a lot, simply a formal sentence of the form: "is there any discussion wanted on this PR. If not, I shall merge it tomorrow evening". Otherwise, we shall have the worse of #5986.

AFAIK, we never agreed for a minimal 24 hour delay. And I think it doesn't make sense. This PR is a bug fix. You agreed it was a bug in #6998. The implementation is straightforward. So a reduced delay for discussion is fine. On the other hand, when we tried to discuss #268, even after a full day of discussion at Le Croisic with 4 or 5 core developers, it wasn't enough.

> Let's start to do things properly.

AFAICT, things were done properly on this PR.

**Comment by ejgallego** [comment]
> @ejgallego It is a repeated timeout that happened in all recent tests both on CircleCI and Travis.

Yup I mean that we need to check locally if it still builds, and also if it builds faster in 8.7.1 is a regression not a bug in VST.

**Comment by herbelin** [comment]
No, Maxime, we cannot have double standards. We decided to have a true communication between developers and to let this true communication happens, we need to give it time. And the time I'm proposing here is even peanut compared to what we were waiting when you were doing the merges, so there will be only added value to have such a time.

It is just a standard protocol that we have to follow, even if it looks a bit formal, to be sure that everything is made thinking at what the whole set of developers might be interested to say, and not just thinking that because it looks ok for a couple of us, it is necessarily ok for everyone.

There is no time mentioned in the "merging" note, but 1. the question of minimal/maximal time has been asked several times and we shall not escape to eventually address it 2. by definition, for a discussion to happen, it has to be started, and I saw no sign that it started here (even if the change looks natural to me and I'll personally have nothing special more to say than a :+1:).

You're mentioning #268, and of course, not all PRs are comparable in size and complexity. But the need for an explicit place for potential discussions definitely exists in all cases, the simple and the complex ones. If what you are afraid of is that there is a proliferation of messages for formal communication, I'm proposing that in answer to the message of the merge assignee formally asking for "anymore comments?", we simply use the :+1: button to signify that we are ok.

**Comment by maximedenes** [comment]
> the question of minimal/maximal time has been asked several times and we shall not escape to eventually address it

It has been addressed many times in working groups after you raised it, and the answer was, as I understood it, "no fixed minimal delay".

> I'm proposing that in answer to the message of the merge assignee formally asking for "anymore comments?", we simply use the  button to signify that we are ok.

That's too heavy for a daily merging process. Like @ejgallego said, we are already at the limit with the current process that asks all concerned maintainers for approval. Please don't make it even heavier.

To ensure relevant discussion time has been left, please use common sense. You say yourself that there was no problem here, so why exactly are you complaining? Of course, if a PR contains a new feature, more discussion time should be allowed than for a bug fix. And even more time if the feature is complex, even less time if the bug fix is straightforward (like here). If we realize after the fact that a PR was merged without enough discussion, we can always revert it.

I have a strange feeling that this has been said many times already. Could we move on and invest our energy in more productive ways?

Bottom line, @gares, you handled this PR in the way that was expected according to the merge process described in `dev/doc/MERGING.md`. @herbelin, if you're not happy with that, please prepare an amendment to that process that I'll be happy to object against. Reminder: it is completely orthogonal to this PR, so the discussion should continue somewhere else (in a dedicated issue or PR).

**Comment by Zimmi48** [comment]
> Bottom line, @gares, you handled this PR in the way that was expected according to the merge process described in dev/doc/MERGING.md. @herbelin, if you're not happy with that, please prepare an amendment to that process that I'll be happy to object against. Reminder: it is completely orthogonal to this PR, so the discussion should continue somewhere else (in a dedicated issue or PR).

:+1: 
The process was followed as described. If you want to have this delay all the time, open a PR to amend the merging doc. But honestly, I feel that you should first give a few days or a few weeks so that we see how the process is going and we can judge better whether the amendment is a good idea.

**Comment by herbelin** [comment]
> Reminder: it is completely orthogonal to this PR, so the discussion should continue somewhere else (in a dedicated issue or PR).

> The process was followed as described. If you want to have this delay all the time, open a PR to amend the merging doc. 

This PR is the first one implementing the new process and this is why the discussion started here. I opened #7042 for a proper discussion. I did not intend to "hijack" the PR per se (but I'm happy to see, @gares, that you take it with humour).

Maybe the words I used were found too direct (did they?) and if so, I apologize.
